### PR TITLE
External information sources

### PR DESCRIPTION
**Pull request by jhoetter**

### CODE REVIEW

**Comment by jhoetter** [comment]
Database model for PR https://github.com/code-kern-ai/refinery-gateway/pull/39

File: business_objects/embedder.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
+from datetime import datetime
```
Comment (JWittmeyer) [inline business_objects/embedder.py]: is this file part of the change?

File: business_objects/embedder.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    name: Optional[str] = None,
+    type: Optional[str] = None,
+    description: Optional[str] = None,
+    source_code: Optional[str] = None,
+    version: Optional[int] = None,
+    created_at: Optional[datetime] = None,
+    created_by: Optional[str] = None,
+    with_commit: bool = False,
+) -> None:
+    information_source = get(project_id, embedder_id)
```
Comment (JWittmeyer) [inline business_objects/embedder.py]: doesn't seem right (information source)

File: business_objects/embedder_payload.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+from datetime import datetime
```
Comment (JWittmeyer) [inline business_objects/embedder_payload.py]: not part of pr?

File: enums.py [hunk 4]
```diff
@@ -65,6 +66,8 @@ class Tablenames(Enum):
     LABELING_TASK_LABEL = "labeling_task_label"
     EMBEDDING = "embedding"
     EMBEDDING_TENSOR = "embedding_tensor"
+    EMBEDDER = "embedder"
+    EMBEDDER_PAYLOAD = "embedder_payload"
```
Comment (JWittmeyer) [inline enums.py]: new tables without alembic file?

File: business_objects/record_label_association.py [hunk 5]
```diff
@@ -474,6 +474,20 @@ def delete_by_source_id(
...
+def delete_by_source_id_and_record_ids(
+    project_id: str,
+    information_source_id: str,
+    record_ids: List[str],
+    with_commit: bool = False,
+) -> None:
+    session.query(RecordLabelAssociation).filter(
+        RecordLabelAssociation.project_id == project_id,
+        RecordLabelAssociation.source_id == information_source_id,
+        RecordLabelAssociation.record_id.in_(record_ids),
```
Comment (JWittmeyer) [inline business_objects/record_label_association.py:486]: in operator with a list of str isn't scaling proof (batching should help)

File: business_objects/embedder.py (hunk 1 above)
Comment (jhoetter) [reply business_objects/embedder.py]: no, you're right! This was part of https://github.com/code-kern-ai/refinery-submodule-model/pull/6.
I'll delete it in this submodule. Strange that it appeared here

File: business_objects/embedder.py (hunk 2 above)
Comment (jhoetter) [reply business_objects/embedder.py]: deleted the whole file

File: business_objects/embedder_payload.py (hunk 3 above)
Comment (jhoetter) [reply business_objects/embedder_payload.py]: yes

File: enums.py (hunk 4 above)
Comment (jhoetter) [reply enums.py]: again, there was something wrong with the PR. I delete this

File: business_objects/record_label_association.py (hunk 5 above)
Comment (jhoetter) [reply business_objects/record_label_association.py:486]: it is only used in batches of 32, so arguably no need for scaling :)

File: business_objects/information_source.py [hunk 6]
```diff
@@ -168,7 +179,7 @@ def get_exclusion_record_ids_for_task(task_id: str) -> List[str]:
     return exclusion_ids
 
 
-def get_overview_data(project_id: str) -> List[Dict[str, Any]]:
+def get_overview_data(project_id: str, type_selection: str) -> List[Dict[str, Any]]:
```
Comment (JWittmeyer) [inline business_objects/information_source.py]: sorry what i meant was the boolean value and evaluation here so someone using the function knows what to do without checking the SQL code

Comment (jhoetter) [reply business_objects/information_source.py]: makes sense

**Review by JWittmeyer — APPROVED** [review]

**Pull request merged**